hooks/useInfiniteStoriesLoading.ts

```diff
@@ -0,0 +1,75 @@
+import { useCallback, useEffect, useState } from 'react';
+import { Category, Story } from '@prezly/sdk/dist/types';
+import { PaginationProps } from 'types';
+
+async function fetchStories(
+    page: number,
+    pageSize: number,
+    category?: Category,
+): Promise<{ stories: Story[] }> {
+    const result = await fetch('/api/fetch-stories', {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+            page,
+            pageSize,
+            category,
+        }),
+    });
+
+    if (result.status !== 200) {
+        const { message } = await result.json();
+        throw new Error(message);
+    }
+
+    return result.json();
+}
+
+export const useInfiniteStoriesLoading = (
+    initialStories: Story[],
+    pagination: PaginationProps,
+    category?: Category,
+) => {
+    const [displayedStories, setDisplayedStories] = useState<Story[]>(initialStories);
+    const [currentPage, setCurrentPage] = useState(1);
+    const [isLoading, setIsLoading] = useState(false);
+
+    const { itemsTotal, pageSize, currentPage: currentQueryPage } = pagination;
+    const totalPages = Math.ceil(itemsTotal / pageSize);
+
+    // This also disabled infinite loading if query pagination is active
+    const canLoadMore = currentQueryPage === 1 && currentPage < totalPages;
```

Comment:
I don't understand the first part of this condition.

:x: 

But regardless, this is weird. If someone does not want to have infinite loading, they should not use this hook.
So this hook should safely assume that infinite loading is always enabled. 
Also, `useInfiniteStoriesLoading` should know nothing abut query pagination.

I'd rather have us expose 2 components (e.g. `Pagination` & `LoadMore`) for users to pick from, so that there is clear separation of concerns/behavior. I wrote more about it in `Stories` component.